RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.cc

```diff
@@ -0,0 +1,1038 @@
+	    minSum = dsum;  
+	    bad_layer = worst_layer;		   
+	    iworst = irej;  
+	    rh_to_be_deleted_1 = ir; 
+	  } 
+	}//ir  
+      }//i2 
+    }//i1 
+    fitX_ir(sp, se, bad_layer, chi2_str); 
+  }//if chi2prob<1.0e-4 
+
+  //find worst from n-1 hits 
+  int iworst2 = -1; 
+  int bad_layer2 = -1; 
+  if (iworst > -1 && (nhits-1) > n_seg_min && (chi2_str/(nhits-3)) > chi2_str_*chi2D_iadd){ 
+    iworst = -1; 
+    float minSum = 1000; 
+    int i1b = 0; 
+    int i2b = 0;  
+    for (ChamberHitContainer::iterator i1 = proto_segment.begin(); i1 != proto_segment.end();++i1) { 
+      ++i1b;  
+      const CSCRecHit2D* i1_1 = *i1; 
+      CSCDetId idRH1 = (CSCDetId)(*i1_1).cscDetId(); 
+      int z1 = idRH1.layer(); 
+      i2b = i1b; 
+      for ( ChamberHitContainer::iterator i2 = i1+1; i2 != proto_segment.end(); ++i2) { 
+	++i2b;  
+	const CSCRecHit2D* i2_1 = *i2; 
+
+	CSCDetId idRH2 = (CSCDetId)(*i2_1).cscDetId(); 
+	int z2 = idRH2.layer(); 
+	int irej = 0; 
+
+	for ( ChamberHitContainer::iterator ir = proto_segment.begin(); ir != proto_segment.end(); ++ir) { 
+
+	  ++irej;   
+	  int irej2 = 0; 
+	  if (ir == i1 || ir == i2 ) continue;  
+	  const CSCRecHit2D* ir_1 = *ir; 
+	  CSCDetId idRH = (CSCDetId)(*ir_1).cscDetId(); 
+	  int worst_layer = idRH.layer(); 
+	  for (  ChamberHitContainer::iterator ir2 = proto_segment.begin(); ir2 != proto_segment.end(); ++ir2) { 
+
+            ++irej2;   
+	    if (ir2 == i1 || ir2 == i2 || ir2 ==ir ) continue;  
+	    float dsum = 0; 
+	    int hit_nr = 0; 
+	    const CSCRecHit2D* ir2_1 = *ir2; 
+	    CSCDetId idRH = (CSCDetId)(*ir2_1).cscDetId(); 
+	    int worst_layer2 = idRH.layer(); 
+	    for (  ChamberHitContainer::iterator i = proto_segment.begin(); i != proto_segment.end(); ++i) {  
+	      ++hit_nr;  
+	      const CSCRecHit2D* i_1 = *i; 
+	      if (i == i1 || i == i2 || i == ir|| i == ir2 ) continue;  
+	      float slope = (sp(z2-1)-sp(z1-1))/(z2-z1); 
+	      float intersept = sp(z1-1) - slope*z1; 
+	      CSCDetId idRH = (CSCDetId)(*i_1).cscDetId(); 
+	      int z = idRH.layer(); 
+	      float di = fabs(sp(z-1) - intersept - slope*z); 
+	      dsum = dsum + di; 
+	    }//i 
+	    if (dsum < minSum){ 
+	      minSum = dsum;  
+	      iworst2 = irej2;  
+	      iworst = irej; 
+	      bad_layer = worst_layer; 
+	      bad_layer2 = worst_layer2;  
+	      rh_to_be_deleted_1 = ir; 
+	      rh_to_be_deleted_2 = ir2; 
+	    } 
+	  }//ir2  
+	}//ir  
+      }//i2 
+    }//i1	 
+
+    fitX_ir2(sp, se, bad_layer ,bad_layer2, chi2_str); 
+  }//if prob(n-1)<e-4 
+
+  //---------------------------------- 
+  //erase bad_hits 
+  //---------------------------------- 
+
+  if( iworst2-1 >= 0 && iworst2 <= int(proto_segment.size())  ) { 
+    proto_segment.erase( rh_to_be_deleted_2); 
+  } 
+
+  if( iworst-1 >= 0 && iworst <= int(proto_segment.size())  ){ 
+    proto_segment.erase(rh_to_be_deleted_1); 
+  } 
+} 
+
+float CSCSegAlgoRU::fit_sp(SVector6 points, SVector6 errors, int layer){ 
+

```

Comment:
fit_sp() function used previously for validation of the code and not used anymore. To be deleted.